standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```
@@ -849,8 +893,10 @@ void updateCreationMetadata(String catName, String dbName, String tableName, Cre
    * @throws MetaException error accessing the RDBMS
    * @throws TException thrift transport error
    */
-  Partition appendPartition(String dbName, String tableName, List<String> partVals)
-      throws InvalidObjectException, AlreadyExistsException, MetaException, TException;
+  default Partition appendPartition(String dbName, String tableName, List<String> partVals)
```

should throw UnsupportedOperationException

---

Done.